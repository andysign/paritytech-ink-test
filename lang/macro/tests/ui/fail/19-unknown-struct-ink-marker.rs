use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    #[ink(unknown_or_unsupported)]
    struct HasUnknownMarker {}

    impl Noop {
        #[ink(constructor)]
        fn new(&mut self) {}

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
