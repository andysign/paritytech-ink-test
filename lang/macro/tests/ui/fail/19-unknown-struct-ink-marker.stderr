error: encountered unsupported ink! markers for struct
 --> $DIR/19-unknown-struct-ink-marker.rs:8:5
  |
8 | /     #[ink(unknown_or_unsupported)]
9 | |     struct HasUnknownMarker {}
  | |______________________________^

error: unsupported ink! marker for struct
 --> $DIR/19-unknown-struct-ink-marker.rs:8:10
  |
8 |     #[ink(unknown_or_unsupported)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^
